lib/rules/no-console.js

```diff
@@ -122,6 +122,9 @@ module.exports = {
                 if (!shadowed) {
                     references
                         .filter(isMemberAccessExceptAllowed)
+
+                        // only report errors on CallExpression
+                        .filter(item => item.identifier.parent.parent.type === "CallExpression")
```

Comment:
This will report errors when `console.log` is an argument:

```js
foo(console.log);
```

Based on https://github.com/eslint/eslint/issues/7806#issuecomment-269565002 I think that should not report an error.

This could be fixed by doing something like:

```diff
-    .filter(item => item.identifier.parent.parent.type === "CallExpression")
+    .filter(item => item.identifier.parent.parent.type === "CallExpression" && item.identifier.parent === item.identifier.parent.parent.callee)
```